tests/test_app.py

```diff
@@ -140,6 +162,127 @@ def test_app_init_with_invalid_name(self, name):
+                        config_json='firebase_config_partial.json',
+                        init_options=None,
+                        want_options={'databaseURL': 'https://hipster-chat.firebaseio.mock',
+                                      'projectId': 'hipster-chat-mock'}),
+        OptionsTestCase(name='no options, partial config in json',
+                        config_json='{"databaseURL": "https://hipster-chat.firebaseio.mock",' +
+                        '"projectId": "hipster-chat-mock"}',
+                        init_options=None,
+                        want_options={'databaseURL': 'https://hipster-chat.firebaseio.mock',
+                                      'projectId': 'hipster-chat-mock'}),
+        OptionsTestCase(name='partial config file is ignored',
+                        config_json='firebase_config_partial.json',
+                        init_options={'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-mock'},
+                        want_options={'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-mock'}),
+        OptionsTestCase(name='full config file is ignored',
+                        config_json='firebase_config.json',
+                        init_options={'databaseAuthVariableOverride': 'davy1-mock',
+                                      'databaseURL': 'https://db1-mock',
+                                      'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-.mock'},
+                        want_options={'databaseAuthVariableOverride': 'davy1-mock',
+                                      'databaseURL': 'https://db1-mock',
+                                      'projectId': 'pid1-mock',
+                                      'storageBucket': 'sb1-.mock'}),
+        OptionsTestCase(name='full config file is ignored with missing values in options',
+                        config_json='firebase_config.json',
+                        init_options={'databaseAuthVariableOverride': 'davy1 - mock',
+                                      'projectId': 'pid1 - mock',
+                                      'storageBucket': 'sb1 - .mock'},
+                        want_options={'databaseAuthVariableOverride': 'davy1 - mock',
+                                      'projectId': 'pid1 - mock',
+                                      'storageBucket': 'sb1 - .mock'})]
+
+    @pytest.mark.parametrize('test_case', options_test_cases)
+    def test_app_init_with_default_config(self, test_case):
+        """Set the CONFIG env var and test that options are initialized"""
+        config_old = set_config_env(test_case.config_json)
+        app = firebase_admin.initialize_app(CREDENTIAL, options=test_case.init_options)
+        for field in firebase_admin._CONFIG_VALID_KEYS:
+            assert app.options.get(field) == test_case.want_options.get(field), test_case.name
+        revert_config_env(config_old)
```

Comment:
This won't run if an assertion fails.  See if you can turn it into a self-cleaning fixture like https://github.com/firebase/firebase-admin-python/blob/master/tests/test_credentials.py#L83
  
If that's not possible, at least use a try-finally block.

Reply:
I've put it in the fixture according to your first suggestion, thanks. 
PTAL